### PR TITLE
Add: set_max_message_length, closes #31

### LINKED ISSUE #31: Expose the max message size to user.
The max grpc message size is limited to [500MB](https://github.com/ray-project/rayfed/blob/main/fed/_private/grpc_options.py#L27) by now, it's better to expose these options to user.

From https://github.com/secretflow/secretflow/issues/294

### PR DESCRIPTION
**Pull request by Krout0n**
I have confirmed this PR working by opening an interacted-shell as shown in the screenshot below. closes #31 

スクリーンショット 2023-02-04 23 18 34

### CODE REVIEW

**Review by jovany-wang — COMMENTED** [review]
So do we need a unit test for this? like you could just add what you wrote in your PR description as a unit test.

File: fed/_private/grpc_options.py [hunk 1]
```diff
@@ -24,9 +24,23 @@
...
-_GRPC_MAX_RECEIVE_MESSAGE_LENGTH = 500 * 1024 * 1024
+_DEFAULT_GRPC_MAX_SEND_MESSAGE_LENGTH = 500 * 1024 * 1024
+_DEFAULT_GRPC_MAX_RECEIVE_MESSAGE_LENGTH = 500 * 1024 * 1024
 
+_GRPC_MAX_SEND_MESSAGE_LENGTH = _DEFAULT_GRPC_MAX_SEND_MESSAGE_LENGTH
+_GRPC_MAX_RECEIVE_MESSAGE_LENGTH = _DEFAULT_GRPC_MAX_RECEIVE_MESSAGE_LENGTH
+
+def set_max_message_length(new_max_size):
+    """
+    Set the maximum length of gRPC messages.
```
[the remark this answers is not in the document]
Comment (jovany-wang) [reply fed/_private/grpc_options.py:35]: @zhouaihui Do we need to add the code style check to CI?

File: fed/api.py [hunk 2]
```diff
@@ -52,6 +53,7 @@ def init(
     cross_silo_send_max_retries: int = None,
     cross_silo_serializing_allowed_list: Dict = None,
     exit_on_failure_cross_silo_sending: bool = False,
+    grpc_max_size_in_bytes: int = None,
```
Comment (zhouaihui) [inline fed/api.py:56]: `grpc_max_size_in_bytes` -> `cross_silo_message_max_size_in_bytes` could be more explicit?

**Review by zhouaihui — CHANGES_REQUESTED** [review]

**Comment by zhouaihui** [comment]
Both patterns are good to me. But the failure case is not easy to catch, you can have a try 👍🏻.

File: fed/api.py (hunk 2 above)
Comment (Krout0n) [reply fed/api.py:56]: Sounds good to me@

**Comment by jovany-wang** [comment]
> If you don't need the unit test to merge this PR, just submit an issue and remind me after a few days if I can do it 😆

@Krout0n OK. That sounds good as this PR is simple. 
@zhouaihui Maybe we could merge this PR first, and then create an issue for the coming unit test.

**Comment by zhouaihui** [comment]
> Maybe we could merge this PR first, and then create an issue for the coming unit test.

+1. LGTM

**Review by zhouaihui — APPROVED** [review]
LGTM

**Review by jovany-wang — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**